TeachingRecordSystem/src/TeachingRecordSystem.Core/Jobs/EWCWalesImportJob.cs

```diff
@@ -0,0 +1,204 @@
+                        var subfolderFileNames = await GetFileNamesAsync(containerClient, blobhierarchyItem.Prefix, true, cancellationToken);
+                        if (subfolderFileNames != null)
+                        {
+                            fileNames.AddRange(subfolderFileNames);
+                        }
+                    }
+                }
+                else
+                {
+                    fileNames.Add(blobhierarchyItem.Blob.Name);
+                }
+            }
+        }
+
+        return fileNames.ToArray();
+    }
+
+    public async Task<MemoryStream> GetDownloadStreamAsync(string fileName)
+    {
+        BlobContainerClient containerClient = blobServiceClient.GetBlobContainerClient(_storageContainer);
+        BlobClient blobClient = containerClient.GetBlobClient($"{fileName}");
+        var downloadStream = new MemoryStream();
+        await blobClient.DownloadToAsync(downloadStream);
+        downloadStream.Position = 0;
+        return downloadStream;
+    }
+
+    public async Task<(int TotalCount, int SuccessCount, int DuplicateCount, int FailureCount, string FailureMessage, Guid? IntegrationTransactionId)> ImportAsync(string fileName, StreamReader reader)
+    {
+        var fileNameParts = fileName.Split("/");
+        var fileNameWithoutFolder = fileNameParts.Last();
+        var integrationJob = new CreateIntegrationTransactionQuery()
+        {
+            StartDate = DateTime.Now,
+            TypeId = (int)dfeta_IntegrationInterface.GTCWalesImport,
+            FileName = fileName
+        };
+
+        var integrationId = await crmQueryDispatcher.ExecuteQueryAsync(integrationJob);
+        TryGetImportFileType(fileNameWithoutFolder, out var importType);
+        var (totalCount, successCount, duplicateCount, failureCount, failureMessage) = (0, 0, 0, 0, "");
+
+        if (importType == EwcWalesImportFileType.Induction)
+        {
+            var results = await inductionImporter.ImportAsync(reader, integrationId, fileNameWithoutFolder);
+            totalCount = results.TotalCount;
+            successCount = results.SuccessCount;
+            duplicateCount = results.DuplicateCount;
+            failureCount = results.FailureCount;
+            failureMessage = results.FailureMessage;
+        }
+        else if (importType == EwcWalesImportFileType.Qualification)
+        {
+            var results = await qtsImporter.ImportAsync(reader, integrationId, fileNameWithoutFolder);
+            totalCount = results.TotalCount;
+            successCount = results.SuccessCount;
+            duplicateCount = results.DuplicateCount;
+            failureCount = results.FailureCount;
+            failureMessage = results.FailureMessage;
+        }
+        else
+        {
+            totalCount = 0;
+            successCount = 0;
+            duplicateCount = 0;
+            failureCount = 0;
+            failureMessage = "Import filename must begin with IND or QTS";
```

Comment:
This could probably be an exception rather than putting it into the Integration transaction.